engine/server/engine/streaming/websocket_pump.go

```diff
@@ -0,0 +1,119 @@
+package streaming
+
+import (
+	"context"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/labstack/echo/v4"
+	"github.com/rs/cors"
+	"github.com/sirupsen/logrus"
+
+	api_type "github.com/kurtosis-tech/kurtosis/api/golang/http_rest/api_types"
+)
+
+const (
+	wsReadBufferSize  = 1024
+	wsWriteBufferSize = 1024
+	maxMessageSize    = 512
+	// Time allowed to write a message to the peer.
+	writeWait = 10 * time.Second
+	// Time allowed to read the next pong message from the peer.
+	pongWait = 60 * time.Second
+	// Send pings to peer with this period. Must be less than pongWait.
+	// nolint:gomnd
+	pingPeriod = (pongWait * 9) / 10
+)
+
+type WebsocketPump[T interface{}] struct {
+	websocket  *websocket.Conn
+	inputChan  chan *T
+	infoChan   chan *api_type.ResponseInfo
+	ctx        context.Context
+	cancelFunc context.CancelFunc
+}
+
+func NewWebsocketPump[T interface{}](ctx echo.Context, cors cors.Cors) (*WebsocketPump[T], error) {
+	// nolint: exhaustruct
+	upgrader := websocket.Upgrader{
+		ReadBufferSize:  wsReadBufferSize,
+		WriteBufferSize: wsWriteBufferSize,
+		CheckOrigin:     cors.OriginAllowed,
+	}
+
+	conn, err := upgrader.Upgrade(ctx.Response(), ctx.Request(), nil)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to upgrade http connection to websocket")
+	}
+
+	ctxWithCancel, cancelFunc := context.WithCancel(context.Background())
+
+	return &WebsocketPump[T]{
+		websocket:  conn,
+		inputChan:  make(chan *T),
+		ctx:        ctxWithCancel,
+		cancelFunc: cancelFunc,
+	}, nil
+}
+
+func (pump WebsocketPump[T]) StartPumping() {
+	ticker := time.NewTicker(pingPeriod)
+	defer func() {
+		ticker.Stop()
+		pump.websocket.Close()
+	}()
+
+	logrus.WithFields(logrus.Fields{
+		"pongWait":       pongWait,
```

Comment:
nit: you could use constants for these keys

Reply:
what's the benefity in this case? it does seems an overkill, no?

Reply:
probably is an over-engineering suggestion, I use to do it for me or someone else copy the code later both pieces have the same key and then if it is edited, will be edited in only one place, but.
Not necessary to do the change.